app/src/main/java/com/example/barterbuddy/activities/BarterPage.java

```diff
@@ -36,16 +72,127 @@ protected void onCreate(Bundle savedInstanceState) {
 
     getXmlElements();
 
-    // listener for abort trade button, closes trade and starts "tinder" activity
+    loadPage();
+  }
+
+  private void loadPage() {
+    getTradeData();
+  }
+
+  private void getTradeData() {
+    String fieldName;
+    if (isPoster) {
+      fieldName = "posterEmail";
+    } else {
+      fieldName = "offeringEmail";
+    }
+    DB.collection("trades")
+        .whereEqualTo(fieldName, email)
+        .whereEqualTo("stateOfCompletion", "BARTERING")
+        .get()
+        .addOnCompleteListener(
```

Comment:
I prefer one on success and one on failure listener rather than an on complete listener with an if statement, but that's just a personal preference and not a big deal.

Reply:
OnFailureListener added.